lib/view/monthlyTaskView.dart

```diff
@@ -1,60 +1,96 @@
 import 'package:flutter/material.dart';
+import 'package:get/get.dart';
 import 'package:planner/common/database.dart';
-import 'package:planner/common/time_management.dart';
 import 'package:planner/models/task.dart';
 import 'package:table_calendar/table_calendar.dart';
 import 'package:planner/view/taskView.dart';
 
 class MonthlyTaskView extends StatefulWidget {
-  const MonthlyTaskView({super.key});
-
   @override
   _MonthlyTaskViewState createState() => _MonthlyTaskViewState();
 }
 
 class _MonthlyTaskViewState extends State<MonthlyTaskView> {
-  CalendarFormat _calendarFormat = CalendarFormat.month;
+  final CalendarFormat _calendarFormat = CalendarFormat.month;
   DateTime _focusedDay = DateTime.now();
   DateTime? _selectedDay;
   DatabaseService db = DatabaseService();
   List<Task> todayTasks = [];
-
-  // Add a PageController for handling page navigation
-   final PageController _pageController = PageController();
+  final PageController _pageController = PageController();
+  Map<DateTime, List<Task>> active = {};
 
   @override
   void initState() {
     super.initState();
-    // Add a listener to the PageController to update the focusedDay
-    _pageController.addListener(() {
-      setState(() {
-        _focusedDay = _pageController.page == 0
-            ? _focusedDay = getDateOnly(_focusedDay, offsetMonths: -1)
-            : _focusedDay = getDateOnly(_focusedDay, offsetMonths: 1);
-      });
-    });
+    fetchMonthlyTasks(DateTime.now());
   }
 
   @override
   void dispose() {
-    // Dispose of the PageController to prevent memory leaks
     _pageController.dispose();
     super.dispose();
   }
 
+  bool hasTasks(
```

Comment:
This functions feels like it could be a static function as well. Maybe it shouldn't be in this class and moved out somewhere. Perhaps a file with a bunch of helper functions in lib/common?

Reply:
feels like it violates one of the SOLID principles

Reply:
should definitely be pulled out of here IMO